electricpy/__init__.py

```diff
@@ -6573,9 +6573,277 @@ def wireresistance(length=None,diameter=None,rho=16.8*10**-9,R=None):
         A = rho*length/R
         return _np.sqrt(4*A/pi)
         
+def ic_555(mode,R=None,C=None,f=None,t_high=None,t_low=None):
+
+    if mode == 'ASTABLE':
+
+        if R!=None and C!=None:
+
+            try:
+                assert len(R) == 2
+            except AssertionError:
+                raise ValueError(f"ic_555 in {mode} will have only 2 resitances to be fixed but {len(R)} were \
+                                given")
+
+            [R1, R2] = R
+
+            T = _np.log(2)*C*(R1+2*R2)
+            f = 1/T
+            t_low = _np.log(2)*C*R2
+            t_high = _np.log(2)*C*(R1+R2)
+            duty_cycle = t_high*100/T
+
+            return {
+                'time_period':T,
+                'frequency':f,
+                'duty_cycle':duty_cycle,
+                't_low':t_low,
+                't_high':t_high
+            }
+        
+        elif t_high!=None and t_low!=None and C!=None:
+
+            x2 = t_low/C*_np.log(2)
+            x1 = t_high/C*_np.log(2)
+
+            T = t_high+t_low
+
+            f = 1/(T)
+
+            duty_cycle = t_high/(T)
+
+            return {
+                'time_period':T,
+                'frequency':f,
+                'duty_cycle':duty_cycle,
+                'R1':x1-x2,
+                'R2':x2
+            }
+        else:
+            raise TypeError("Not enough parqmeters are passed")
+            
+              
+    elif mode=='MONOSTABLE':
+        if R is None:
+            try:
+                assert C!=None and T!=None
+            except AssertionError:
+                raise ValueError("To find Resitance, Capacitance and delay time should be provided")
+            return T/(_np.log(3)*C)
+        elif C is None:
+            try:
+                assert R!=None and T!=None
+            except AssertionError:
+                raise ValueError("To find Capacitance , Resistance and delay time should be provided")
+            return T/(_np.log(3)*R)
+
+        elif T is None:
+
+            try:
+                assert R!=None and T!=None
+            except AssertionError:
+                raise ValueError("To find Time delay , Resistance and Capacitance should be provided")
+            return R*C*_np.log(3)
+
+    else:
+        raise ValueError("ic_555 is configured only in `MONOSTABLE` and `ASTABLE` modes")
+
+        
+class InductionMotorCircle:
```

Comment:
I'm thoroughly impressed by this class, and I'm really appreciative that you spent the time to refactor it as a class instead of several functions. Still, there's no documentation on what the class is, or how it should be used. We will need to have one class-level docstring to illustrate what the class is, and how it should be used. It would also be very useful to have some comments and (internal) method-level docstrings explaining what each method is intended to do.

Once we add the documentation strings, I'll be very happy to add this to ElectricPy!

Reply:
Yes I will definitely add doc strings